### PR TITLE
Joe/tour guide

### PR DESCRIPTION
**Pull request by jcrotty**

### CODE REVIEW

File: server/model/user/user.model.server.js [hunk 1]
```diff
@@ -32,6 +36,18 @@ function updateUser(userId, user) {
...
+function addTourGuideArea(userId, newArea) {
+  return Address.createAddress(newArea)
+    .then(function (newAddress) {
+      return UserModel.findById(userId)
+        .then(function(user) {
+          user.tourGuideAreas.push(newAddress._id);
+          user.save();
+          return user;
+        })
+    })
```
Comment (alfred) [inline server/model/user/user.model.server.js]: Pls follow the non-inner Promise pattern
```js
let newAddress = new Address( newArea );

return UserModel.findById( userId )
  .then( user => { 
    user.tourGuideAreas.push( newAddress._id );
    return user.save();
  })
  .then( () => newAddress.save() );
```

File: public/views/tour-guide/tour-guide.controller.client.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      UserService.findUserById(userId)
+        .success(function (user) {
+          vm.user = user;
+          vm.address = user.tourGuideAreas.find(area => area._id === addressId);
+          console.log(vm.address);
+        })
+        .error(function (err) {
+          vm.error = "Error retrieving user.";
+          console.log("error:", err);
+        });
```
Comment (alfred) [inline public/views/tour-guide/tour-guide.controller.client.js]: Use the `$rootScope.currentUser` there is no need to make this call at all

File: public/views/tour-guide/tour-guide.controller.client.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    function init() {
+      UserService.findUserById(userId)
+        .success(function (user) {
+          vm.user = user;
+        })
+        .error(function (err) {
+          vm.error = "Error retrieving user.";
+          console.log("error:", err);
+        });
```
Comment (alfred) [inline public/views/tour-guide/tour-guide.controller.client.js]: Seriously, stop doing this.

File: public/views/tour-guide/tour-guide.controller.client.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    var vm = this;
+
+    vm.title = $route.current.$$route.title;
+    var userId = $rootScope.currentUser && $rootScope.currentUser._id;
+    vm.noBackButton = true;
+
+    function init() {
+      TripService.findAllTripsByTourGuide(userId)
+        .success(function (guides) {
+          vm.guides = guides;
```
Comment (alfred) [inline public/views/tour-guide/tour-guide.controller.client.js]: This function doesn't return `guides` it returns `trips` so I have no clue what you mean

File: public/views/tour-guide/tour-guide.controller.client.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      });
+    }
+    init();
+  }
+
+  function TourGuideViewController(TripService, UserService, $routeParams, $rootScope, $route) {
+    var vm = this;
+
+    vm.title = $route.current.$$route.title;
+    vm.user = vm.userId = $rootScope.currentUser;
```
Comment (alfred) [inline public/views/tour-guide/tour-guide.controller.client.js]: Re-read this.

File: public/services/user.service.client.js [hunk 6]
```diff
@@ -34,10 +36,20 @@
       return $http.get("/api/user?username=" + username + "&password=" + password);
     }
 
+    function findTourGuidesForAddress(address) {
+      let params = "city=" + address.city + "&state=" + address.state + "&country=" + address.country;
+
+      return $http.get("/api/tourGuidesForAddress?" + params);
```
Comment (jcrotty) [inline public/services/user.service.client.js]: I tried to make this `/api/user/tourGuidesForAddress` but couldn't make it work I kept getting 404 Im probably just dumb though someone send help

File: public/services/user.service.client.js [hunk 7]
```diff
@@ -34,10 +36,20 @@
       return $http.get("/api/user?username=" + username + "&password=" + password);
     }
 
+    function findTourGuidesForAddress(address) {
+      let params = "city=" + address.city + "&state=" + address.state + "&country=" + address.country;
+
+      return $http.get("/api/user/tourGuidesForAddress?" + params);
```
Comment (alfred) [inline public/services/user.service.client.js]: I don't get why you guys hate using the expanded `$http`

File: public/views/trip/trip-view/trip-view.controller.client.js [hunk 8]
```diff
@@ -47,7 +49,16 @@
       }
 
     }
-
     init();
+
+    function deleteTourGuide() {
+      console.log("deleteTourGuide");
```
Comment (alfred) [inline public/views/trip/trip-view/trip-view.controller.client.js]: Lol

File: server/model/trip/trip.model.server.js [hunk 9]
```diff
@@ -36,7 +37,7 @@ function findTripById( tripId ) {
 };
 
 function findTripsByTourGuide( tourGuideId ) {
-  return Trip.find( { tourGuide: tourGuideId } );
+  return Trip.find( { tourGuide: tourGuideId } ).populate("_owner");
```
Comment (alfred) [inline server/model/trip/trip.model.server.js]: @biross Doesn't he have to call `.exec()`?

File: server/model/user/user.model.server.js [hunk 10]
```diff
@@ -30,13 +35,40 @@ function findUserByCredentials(username, password) {
...
+function findTourGuidesForAddress(city, state, country) {
+  return UserModel.find( {type: 'USER-TOUR-GUIDE'} )
+    .populate( 'tourGuideAreas' )
+    .then( tourGuides => {
+      return tourGuides.filter( guide => {
+        var matchingAreas = guide.tourGuideAreas.filter( address => {
+            return (!city || address.city === city)
+                && (!state || address.state === state)
+                && (!country || address.country === country);
+          });
```
Comment (alfred) [inline server/model/user/user.model.server.js]: This anonymous function should come out and be named and be a private function (so just don't export it).

File: server/model/user/user.model.server.js [hunk 11]
```diff
@@ -30,13 +35,40 @@ function findUserByCredentials(username, password) {
...
+
+        return matchingAreas.length > 0;
```
Comment (alfred) [inline server/model/user/user.model.server.js]: Just return `matchingAreas.length`

File: server/model/trip/trip.model.server.js (hunk 9 above)
Comment (alfred) [inline server/model/trip/trip.model.server.js]: Sorry @Omniforce

Comment (Omniforce) [inline server/model/trip/trip.model.server.js]: This is something I still don't get. Its something like without `.exec()` it returns a Mongoose Query, while `.exec()` returns a real promise... but they both support `.then()` so it doesn't really matter

Comment (alfred) [inline server/model/trip/trip.model.server.js]: Weird API rules, okay I'll let this slide then

File: server/model/user/user.model.server.js (hunk 10 above)
Comment (alfred) [inline server/model/user/user.model.server.js]: Nah fuck this, just style it nicely

**Pull request merged**